src/features/codeActionProvider.ts

```diff
@@ -9,34 +9,26 @@ import AbstractProvider from './abstractProvider';
 import * as protocol from '../omnisharp/protocol';
 import { toRange2 } from '../omnisharp/typeConvertion';
 import * as serverUtils from '../omnisharp/utils';
-import { Options } from '../omnisharp/options';
 import { FileModificationType } from '../omnisharp/protocol';
 import { Uri } from 'vscode';
 import CompositeDisposable from '../CompositeDisposable';
+import OptionStream from '../observables/OptionStream';
 
 export default class CodeActionProvider extends AbstractProvider implements vscode.CodeActionProvider {
 
-    private _options: Options;
     private _commandId: string;
 
-    constructor(server: OmniSharpServer) {
+    constructor(server: OmniSharpServer, private optionStream: OptionStream) {
         super(server);
 
         this._commandId = 'omnisharp.runCodeAction';
-
-        this._resetCachedOptions();
-
-        let d1 = vscode.workspace.onDidChangeConfiguration(this._resetCachedOptions, this);
         let d2 = vscode.commands.registerCommand(this._commandId, this._runCodeAction, this);
-        this.addDisposables(new CompositeDisposable(d1, d2));
-    }
-
-    private _resetCachedOptions(): void {
-        this._options = Options.Read(vscode);
+        this.addDisposables(new CompositeDisposable(d2));
```

Comment:
Since there's just a single `Disposable`, does it need to be wrapped in a `CompositeDisposable`?

Reply:
The addDisposables takes a CompositeDisposable as an argument